dace/sdfg/replace.py

```diff
@@ -67,6 +71,46 @@ def replace_dict(subgraph: 'StateSubgraphView',
         for symname, new_name in repl.items()
     }
 
+    # Replace AccessNode with tasklet with constant value
+    sdfg = subgraph.sdfg
+    if isinstance(subgraph, dace.SDFGState):
+        state = subgraph
+        for node in subgraph.nodes():
+            # Make sure to replace only Scalar AccessNodes with no incoming edges
+            if isinstance(node, nodes.AccessNode):
+                if node.data in state.sdfg.arrays:
+                    desc = node.desc(state)
+                # In case the AccessNode name was replaced in the sdfg.arrays but not in the SDFG itself
+                # then we have to look for the replaced value in the sdfg.arrays
+                elif repl[node.data] in state.sdfg.arrays:
+                    desc = state.sdfg.arrays[repl[node.data]]
+                else:
+                    continue
+                if state.in_degree(node) == 0 and not desc.transient and isinstance(desc, data.Scalar):
+                    node_data_symbolic = dace.symbolic.pystr_to_symbolic(node.data)
+                    if node_data_symbolic in symrepl:
+                        tasklet = state.add_tasklet(name="constant",
+                                                    inputs={},
+                                                    outputs={f'{node.data}_value'},
+                                                    code=f'{node.data}_value = {symrepl[node_data_symbolic]}')
+                        access_node_name, _ = sdfg.add_transient(f'{node.data}', [1],
```

Comment:
For reference
```
                                                                 type(sympyexpr_to_python(symrepl[node_data_symbolic])),
```
was necessary because `symbolic.evaluate` didn't always return an `int` but it could return a `symbolic.Number.Integer` which was problematic with `sdfg.add_transient`